osquery/events/windows/windows_event_log.h

```diff
@@ -0,0 +1,118 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#define _WIN32_DCOM
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#include <winevt.h>
+
+#include <osquery/events.h>
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+
+/**
+ * @brief Subscription details for Windows Event Logs
+ *
+ * This context is specific to the Windows Event Log publisher.
+ * Subscribers can pass a vector of source values indicating which
+ * Windows event logs the subscriber wishes to subscribe to.
+ */
+struct WindowsEventLogSubscriptionContext : public SubscriptionContext {
+  /// Channel or Path of the Windows Event Log to subscribe to
```

Comment:
Not sure I follow what you mean here. Are you saying I need a `public:` version of the `friend class`?